src/test/integration/org/apache/cassandra/sidecar/routes/BaseTokenRangeIntegrationTest.java

```diff
@@ -0,0 +1,257 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.routes;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Range;
+
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.cassandra.distributed.UpgradeableCluster;
+import org.apache.cassandra.distributed.api.IInstanceConfig;
+import org.apache.cassandra.distributed.api.TokenSupplier;
+import org.apache.cassandra.sidecar.IntegrationTestBase;
+import org.apache.cassandra.sidecar.adapters.base.Partitioner;
+import org.apache.cassandra.sidecar.common.data.TokenRangeReplicasResponse;
+import org.apache.cassandra.testing.AbstractCassandraTestContext;
+import org.apache.cassandra.testing.CassandraIntegrationTest;
+import org.apache.cassandra.testing.ConfigurableCassandraTestContext;
+
+import static org.apache.cassandra.distributed.shared.NetworkTopology.dcAndRack;
+import static org.apache.cassandra.distributed.shared.NetworkTopology.networkTopology;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test the token range replica mapping endpoint with cassandra container.
+ */
+public class BaseTokenRangeIntegrationTest extends IntegrationTestBase
+{
+
+    protected void validateTokenRanges(TokenRangeReplicasResponse mappingsResponse,
+                                       List<Range<BigInteger>> expectedRanges)
+    {
+        List<TokenRangeReplicasResponse.ReplicaInfo> writeReplicaSet = mappingsResponse.writeReplicas();
+        List<TokenRangeReplicasResponse.ReplicaInfo> readReplicaSet = mappingsResponse.readReplicas();
+        List<Range<BigInteger>> writeRanges = writeReplicaSet.stream()
+                                                             .map(r -> Range.openClosed(new BigInteger(r.start()),
+                                                                                        new BigInteger(r.end())))
+                                                             .collect(Collectors.toList());
+
+        List<Range<BigInteger>> readRanges = readReplicaSet.stream()
+                                                           .map(r -> Range.openClosed(new BigInteger(r.start()),
+                                                                                      new BigInteger(r.end())))
+                                                           .collect(Collectors.toList());
+
+
+        assertThat(writeRanges.size()).isEqualTo(writeReplicaSet.size());
```

Comment:
NIT: The size comparison isn't necessary given `containsExactlyElementsOf`"verifies that actual contains exactly the elements of the given iterable and nothing else in the same order."

Reply:
yep, redundant and not required. will remove